pyuvdata/uvdata/uvdata.py

```diff
@@ -3206,6 +3204,19 @@ def check(
         # into the phase_center_catalog.
         self._convert_old_phase_attributes()
 
+        if self.flex_spw_id_array is None:
+            warnings.warn(
+                "flex_spw_id_array is not set. It will be required starting in version "
+                "3.0",
+                DeprecationWarning,
+            )
+        else:
+            # Check that all values in flex_spw_id_array are entries in the spw_array
+            if not np.all(np.isin(self.flex_spw_id_array, self.spw_array)):
+                raise ValueError(
+                    "All values in the flex_spw_id_array must exist in the spw_array."
+                )
+
```

Comment:
Going to reiterate that it would be a way easier transition if a reasonable default could be set in *most* cases.

Reply:
There is a default set when files are read in. Do you think we should also set a default in the check method? I think that most users won't even notice, because they make objects by reading in files.